### PR TITLE
[PRES-348] Refactor coordinator worker to add support for Regisseur scenarios

### PR DESCRIPTION
**Pull request by WVerlaek**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
Adapts the simulated player worker coordinator to allow it to be run on a test worker node when used in a Regisseur scenario, to prevent breaking existing Regisseur scenarios that use simulated players.

#### Release note
REQUIRED: Add a release note to the `##Unreleased` section of CHANGELOG.md. You can find guidance for writing useful release notes [here](../SpatialGDK/Extras/internal-documentation/how-to-write-good-release-notes.md). Documentation changes are exempt from this requirement.

#### Tests
How did you test these changes prior to submitting this pull request?
What automated tests are included in this PR?

STRONGLY SUGGESTED: How can this be verified by QA?

#### Documentation
How is this documented (for example: release note, upgrade guide, feature page, in-code documentation)?

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/WorkerCoordinator/Logger.cs [hunk 1]
```diff
@@ -1,48 +1,65 @@
...
-		public void WriteError(string logMessage, bool logToConsole = false)
+		public void WriteError(string logMessage, bool logToConnectionIfExists = true)
 		{
-			if (logToConsole)
-			{
-				Console.Error.WriteLine(logMessage);
-			}
-			WriteLog("Error: " + logMessage);
+			WriteLogToFile("Error: " + logMessage);
+            if (logToConnectionIfExists && Connection != null)
```
Comment (girayimprobable) [inline SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/WorkerCoordinator/Logger.cs]: Formatting?

Comment (WVerlaek) [reply SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/WorkerCoordinator/Logger.cs]: Fixed, unsure why there were any tabs here

File: SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/WorkerCoordinator/Program.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    /// Executable that connects and runs simulated players.
+    ///
+    /// There are two types of coordinator workers:
+    ///
+    /// ManagedWorkerCoordinator
+    /// Starts as a managed worker and creates its own connection to the hosting deployment (simulated player deployment).
+    /// <seealso cref="ManagedWorkerCoordinator"/>
+    ///
+    /// RegisseurWorkerCoordinator
+    /// Starts as a standalone process, used internally in Improbable for running as part of a Regisseur scenario.
```
Comment (girayimprobable) [inline SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/WorkerCoordinator/Program.cs]:
```suggestion
    /// Starts as a standalone process, only for internal use at Improbable for running as part of a Regisseur scenario.
```

**Comment by girayimprobable** [comment]
We refer to Regisseur as internal only in this code, it should be fine.

**Pull request merged**